NetCord/ChannelType.cs

```diff
@@ -1,18 +1,72 @@
-﻿namespace NetCord;
+namespace NetCord;
 
+/// <summary>
+/// The type of a channel, dictating its features.
+/// </summary>
 public enum ChannelType
 {
+    /// <summary>
+    /// A text channel within a guild.
+    /// </summary>
     TextGuildChannel = 0,
+
+    /// <summary>
+    /// A text channel for direct messages between two users.
+    /// </summary>
     DMChannel = 1,
+
+    /// <summary>
+    /// A voice channel within a guild.
+    /// </summary>
     VoiceGuildChannel = 2,
+
+    /// <summary>
+    /// A text channel for direct messages between multiple users.
+    /// </summary>
     GroupDMChannel = 3,
+
+    /// <summary>
+    /// An organizational category that contains up to 50 channels.
+    /// </summary>
     CategoryChannel = 4,
+
+    /// <summary>
+    /// A channel that users can follow and crosspost into their own guild (formerly news channels).
+    /// </summary>
     AnnouncementGuildChannel = 5,
+
+    /// <summary>
+    /// A temporary sub-channel within an <see cref="AnnouncementGuildChannel"/>.
+    /// </summary>
     AnnouncementGuildThread = 10,
+
+    /// <summary>
+    /// A temporary sub-channel within a <see cref="TextGuildChannel"/> or <see cref="ForumGuildChannel"/>.
```

Comment:
I think all text channels in guilds can contain threads, not only `TextGuildChannel`, maybe you wanted to reference `NetCord.TextGuildChannel` here. Also as mentioned below, `MediaForumGuildChannel` can contain threads as well so 🤷